src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,52 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    if (!Animal.alive.includes(this)) {
+      Animal.alive.push(this);
+    }
+  }
+
+  static removeFromAlive(animal) {
+    Animal.alive = Animal.alive.filter((a) => a !== animal);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
```

Comment:
It would be better to not hard-code the 'hidden' property to always be true, but toggle it instead. This way, the 'hide' method can be used to both hide and unhide the herbivore.